L1Trigger/Phase2L1ParticleFlow/src/L1TCorrelatorLayer1PatternFileWriter.cc

```diff
@@ -0,0 +1,355 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/L1TCorrelatorLayer1PatternFileWriter.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <iostream>
+
+L1TCorrelatorLayer1PatternFileWriter::L1TCorrelatorLayer1PatternFileWriter(const edm::ParameterSet& iConfig,
+                                                                           const l1ct::Event& eventTemplate)
+    : partition_(parsePartition(iConfig.getParameter<std::string>("partition"))),
+      writeInputs_(iConfig.existsAs<std::string>("inputFileName") &&
```

Comment:
These `iConfig.existsAs` could be avoided with a `fillDescription()`. I would change them adding two input parameters (such as `writeInputs` and `writeOutputs`) to actually turn these features on, rather than relying on some other parameter. 

Having a lot of parameters that used conditionally you would simply need to fill them with dummy values when they are not actually used.

Reply:
I do prefer to not have the parameter at all when it's not used, instead of having it filled with dummy values, so that whenever one enables something that requires those parameters but fails to specify them, the code fails cleanly with a 'missing parameter' error (instead of having to rely on the code to check whether the parameter values are dummy or not, which isn't always obvious)

Reply:
Hmm, that will not wash with ConfDB: the top-level parameters of a plugin are treated fixed, and one can not have additional/different parameters depending on the value of some other parameter.
Does this ever need to run as part of the L1T emulator of phase-2 within HLT?

Reply:
This class is only used to produce text files dumps of L1T data to use in board tests - something that one would never want to run in production or in HLT

Reply:
Thanks!